samtranslator/model/eventsources/push.py

```diff
@@ -492,7 +492,7 @@ def _get_permission(self, resources_to_link, stage, suffix):
         if not stage or not suffix:
             raise RuntimeError("Could not add permission to lambda function.")
 
-        path = path.replace('{proxy+}', '*')
+        path = re.sub(r'{([a-zA-Z0-9._-]+|proxy\+)}', '*', path)
```

Comment:
Resource's path part only allow `a-zA-Z0-9._-` and curly braces at the beginning and the end.
<img width="668" alt="2019-06-23 2 19 29" src="https://user-images.githubusercontent.com/4178716/59966901-89542100-955d-11e9-8d0b-16b996506482.png">